controllers/humiocluster_tls.go

```diff
@@ -199,3 +207,63 @@ func constructNodeCertificate(hc *humiov1alpha1.HumioCluster, nodeSuffix string)
 		},
 	}
 }
+
+func (r *HumioClusterReconciler) waitForNewNodeCertificate(ctx context.Context, hc *humiov1alpha1.HumioCluster, expectedCertCount int) error {
+	for i := 0; i < waitForNodeCertificateTimeoutSeconds; i++ {
+		existingNodeCertCount, err := r.nodeCertificateCount(ctx, hc)
+		if err != nil {
+			return err
+		}
+		r.Log.Info(fmt.Sprintf("validating new pod certificate was created. expected pod certificate count %d, current pod certificate count %d", expectedCertCount, existingNodeCertCount))
+		if existingNodeCertCount >= expectedCertCount {
+			return nil
+		}
+		time.Sleep(time.Second * 1)
+	}
+	return fmt.Errorf("timed out waiting to validate new pod certificate was created")
+}
+
+func (r *HumioClusterReconciler) nodeCertificateCount(ctx context.Context, hc *humiov1alpha1.HumioCluster) (int, error) {
```

Comment:
I don't think the method name is good at conveying what it is doing. To me this "count" suffix makes me think that it just returns the number of certificates we have right now, but doesn't touch the resources. This looks to not be true as it does trigger updates of certificates if their hash differs.

Perhaps it would be fine if we'd just add some godoc for it, but I think we should see if we can find a more suitable method name.

Reply:
Right, this was a poor name choice.